### ISSUE TITLE
[Issue] Crooked EMC calculation

### ISSUE THREAD

**Issue by accessdenied0** [sinkillerj/ProjectE#1106]
Hello dear ProjectE-team,

I run in some issues in version 1.9.4, which I find rather strange: Out some reason looses every block or Item its EMC if the recipes includes a recipe with an item (or block of course) with no EMC. That means for example a: If a Piston, which has OD wooden planks who aren't completely EMC registered, looses his EMC. I highly doubt this is intended, so I marked this as issue.

I hope this is reproducible, else I going to post here what you guys need.

access_denied

**Comment by MaPePeR** [comment]
If no wooden plank has an emc value the piston will also not have one. Working as intended, I guess... 
Same thing happens if you remove the emc for cobblestone or Redstone. You can still set the emc for the piston manually. 

I mean: that's the whole point of the emc system, that the values are calculated from the recipe ingredients. And if there is something that does not have an emc value we cannot calculate a value.

**Comment by williewillus** [comment]
I think he/she's saying when there's just *one* ingredient OD name that doesn't have a value the result doesn't have one. Is that correct? @accessdenied0

**Comment by sinkillerj** [comment]
I think @williewillus is spot on, I've been hearing things to that affect from our Japanese players too, something changed in the latest build.

**Comment by accessdenied0** [comment]
@williewillus Yeah you are right. Oak, Dark Oak, Birch, Jungle and more have commonly an EMC. This is not different by me. But I do have forgot to mentioned that It is even enough to have a different recipe registered for one and the same Item/Block without an EMC-value and it doesn't have a value either. 

tl;dr: If anything in a recipe doesn't have an EMC-value, the output doesn't have one either. Doesn't matter if there is a recipe with one or more fully EMC-calculable versions.

**Comment by MaPePeR** [comment]
Hmm. Weird. I will try to write a test for this later. Not sure if it's a bug in the algorithm or wrong data that is given to it.

**Comment by MaPePeR** [comment]
Can not reproduce this. Tried:

* Delete all my configs
* `/projecte setemc 0` with oak planks in hand
* /projecte reloadEMC`
* Piston still has an EMC value.

Any more hints?

**Comment by MaPePeR** [comment]
@accessdenied0 could you enable the mapping dump option in the mapping.cfg and upload/pastebin the generated file?

**Comment by sinkillerj** [comment]
@MaPePeR That wont work as a test case for this bug, add a mod which introduces a wood without a EMC value.

**Comment by MaPePeR** [comment]
That's the test case for "there is an alternative recipe for something (b)  that contains an item without a value(missing)". 

If I remember correctly rubber is not part of the logWood OD. 
We assign a value to all the logWood items.

**Comment by MaPePeR** [comment]
Added MFR to my dev environment: still can not reproduce. All the things that use wood still have an EMC value.  (Rubber Wood has not - but that is expected)

And yes: i made sure that there is no pregen-file.

**Comment by accessdenied0** [comment]
@MaPePeR, the wooden planks were actually only a think-model, but it happens on other items. For example:
* [Glowstone](http://puu.sh/lO0Fz/fbb3f35858.jpg) [Recipe](http://puu.sh/lO2vZ/afdd312a56.jpg) [Recipe](http://puu.sh/lO2Ae/d98be5249a.jpg) [Recipe](http://puu.sh/lO2EN/2585c27778.jpg).
* [Piston](http://puu.sh/lO2KQ/816c1effad.jpg) [Recipe](http://puu.sh/lO2MA/66eee93a6f.jpg) 
I have to admit I do have a quite a lot [mods](http://pastebin.com/V399kxhZ) .

access_denied

**Comment by MaPePeR** [comment]
@accessdenied0 That mappingdump file would really help to figure this out.

In `mapping.cfg` in the `general` category set `B:dumpEverythingToFile=true` and pastebin the file that is generated in the projectE config folder when doing `/projecte reloademc`

**Comment by accessdenied0** [comment]
@MaPePeR, 

[here](http://puu.sh/lO6jv/eb8b888e19.json) I generated it quick

**Comment by MaPePeR** [comment]
Oh. that one is easy: The Piston Recipe is replaced with a custom `IRecipe` that we cannot parse.

**Comment by accessdenied0** [comment]
@MaPePeR,

Yeah I use MT3, but I have a shit ton of blocks who looses their EMC values under them Ender IO for example and I didn't changed anything on them. Also an Example: I didn't added a custom Glowstone or Bookshelf recipe and here are the recipes missing, too.

**Comment by MaPePeR** [comment]
We could parse MineTweaker Recipes before - so maybe that changed with MineTweaker3.
We might need to add code to support MT3

It might be another mod that changes those recipes and replaces them with unparseable ones? - i don't know...

Or does everything work properly, when you downgrade ProjectE? *confused*

**Comment by accessdenied0** [comment]
@MaPePeR,

With version 1.9.2 all is working like it should and most of the recipes have an EMC value (of course IC2, Reika's Rotary and Reactor, some of the TE-stuff and of course everything who has a custom recipe manager (and aren't registered) has none). 

Short: everything who should have a value, have one. 

To your question because a mod who modify recipes: I don't think I have a specific one and if I have one I don't have noticed yet, because it worked perfectly. The thing what the most mods do is to add more recipes or methods to craft things (e.g. to cast Iron with TiC or a method to craft Glowstone (very unused by me) from CloseGaps)

**Comment by MaPePeR** [comment]
Very weird. I can't remember any big changes to that part. 

Could you generate another mapping dump with the older ProjectE version? That would be great.

**Comment by accessdenied0** [comment]
@MaPePeR,

Yes, of course: [here](https://drive.google.com/file/d/0B1BMWHu8n1hXeVpPRXphOTNFZjA/view?usp=sharing) this file is much bigger, I noticed.

**Comment by MaPePeR** [comment]
Much bigger is an understatement... that file is 1.1G big... wtf...

**Comment by MaPePeR** [comment]
I have no idea, what is causing this...
Is this a public pack you are using, so i can get a easy copy of it?

**Comment by accessdenied0** [comment]
@MaPePeR, 

No a private. But honestly: I can't tell either which mod it causes, because I use a giant amount.

355+/- Mods in total. There are a lot of possibilities and causes. But I can paste my Console Log, FMP-log, Java-Version and of course (again) my modlist.

**Comment by MaPePeR** [comment]
Yea, I don't feel like downloading 300+ mods and getting them to work right now ;)

Logs that contain the emc generation for both versions would be highly appreciated! 

Also: thanks for providing all the info. 

And just to make sure, but I don't think it's the case: you didn't disable the irecipe mapper for ore dictionary recipes in the mapping.cfg, right? 
Because there isn't a single recipe with ore dictionary entry's in the dump... 

I suspect something is crashing the crafting mapper, so it stops processing recipes... But I could find a change in the code, yet, that would explain why it worked before. 

There were some big changes between 1.9.4 and 1.9.2,though. We switched from using item id's to item names, for example. But I don't see that causing all the recipes to be gone...

**Comment by accessdenied0** [comment]
@MaPePeR, I have to thank you for your support.

I noticed something strange at the last part, which I didn't noticed before: http://pastebin.com/zsda5kYm (not sure if this is it or useful)

Log 1 with 1.9.4 http://puu.sh/lRCeT/191e8db787.log
Log 2 with 1.9.2 http://puu.sh/lRE3b/9f902c612f.log


And no I didn't changed anything in the configs (except of the "dumpEverythingToFile=true" of course.).

**Comment by MaPePeR** [comment]
This was indeed useful!

```
[16:03:17] [Server thread/FATAL] [ProjectE/ProjectE]: Exception during Mapping Collection from Mapper moze_intel.projecte.emc.mappers.CraftingMapper. PLEASE REPORT THIS! EMC VALUES MIGHT BE INCONSISTENT!
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: java.lang.IllegalArgumentException: Invalid Item with itemName = StopGap:stopgap
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack$NSSItem.<init>(NormalizedSimpleStack.java:115)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack$NSSItem.<init>(NormalizedSimpleStack.java:109)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack.getFor(NormalizedSimpleStack.java:25)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack.getFor(NormalizedSimpleStack.java:56)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack.getFor(NormalizedSimpleStack.java:50)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack.getFor(NormalizedSimpleStack.java:63)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.mappers.CraftingMapper.addMappings(CraftingMapper.java:44)
[16:03:17] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.EMCMapper.map(EMCMapper.java:98)
```

StopGap is the problem here. It registers its items/blocks in a weird way.

**Comment by accessdenied0** [comment]
@MaPePeR, 

thank you a lot for your help. I updated stopgap and it seem to fix most of the issue. The most Items who loosed their EMC where which uses Redstone and Glowstone (Facepalm myself). 

Sadly I can't say this fixed all of them, because for example the Chisel-compat seem to not work. Only the "craft-able" Item has a EMC value, the chiseled ones not. Can this be an issue, too?

**Comment by MaPePeR** [comment]
Yea... there is a similar exception for the Chisel2Mapper in your log, but i could not yet figure out how to solve it.

```
[16:03:16] [Server thread/FATAL] [ProjectE/ProjectE]: Exception during Mapping Collection from Mapper moze_intel.projecte.emc.mappers.Chisel2Mapper. PLEASE REPORT THIS! EMC VALUES MIGHT BE INCONSISTENT!
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: java.lang.NullPointerException
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.registry.GameRegistry$UniqueIdentifier.<init>(GameRegistry.java:389)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.registry.GameData.getUniqueName(GameData.java:226)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.registry.GameRegistry.findUniqueIdentifierFor(GameRegistry.java:447)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.NormalizedSimpleStack.getFor(NormalizedSimpleStack.java:42)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.mappers.Chisel2Mapper.handleCarvingGroup(Chisel2Mapper.java:65)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.mappers.Chisel2Mapper.addMappings(Chisel2Mapper.java:54)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.EMCMapper.map(EMCMapper.java:98)
[16:03:16] [Server thread/INFO] [STDERR/ProjectE]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.PECore.serverStarting(PECore.java:141)
```
Again it fails on getting the unique name for a block - but in a different way, than it failed before...

**Comment by luffykick** [comment]
Hi MaPePeR, 
Like accessdenied0 I've had the same issue with the EMC values disappearing. I have a big pack. 300+ mods so finding the offending mod was a bit of a challenge. I took a crack at solving this and got some extra info that may help solve.

After I removed The Erebus mod the EMC values generated again. I have a suspicion that the issue may be related to the blockid extension mod NotEnoughIDs "http://minecraft.curseforge.com/projects/notenoughids?gameCategorySlug=mc-mods&projectID=235107" When I add a mod that has lots of block/item id's exceeding the natural block/item id limit ProjecE fails to generate most emc values.

Erebus spit out this in the logs:
[05:15:39] [Server thread/FATAL] [ProjectE]: Exception during Mapping Collection from Mapper moze_intel.projecte.emc.mappers.CraftingMapper. PLEASE REPORT THIS! EMC VALUES MIGHT BE INCONSISTENT!
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: java.lang.ArrayIndexOutOfBoundsException: 32767
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at erebus.item.ItemErebusFood.getContainerItem(ItemErebusFood.java:152)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at erebus.item.ItemErebusFood.hasContainerItem(ItemErebusFood.java:169)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.mappers.CraftingMapper.addMappings(CraftingMapper.java:57)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.emc.EMCMapper.map(EMCMapper.java:98)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at moze_intel.projecte.PECore.serverStarting(PECore.java:141)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at java.lang.reflect.Method.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at java.lang.reflect.Method.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at java.lang.reflect.Method.invoke(Unknown Source)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.common.eventbus.EventBus.post(EventBus.java:275)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.Loader.serverStarting(Loader.java:789)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.FMLCommonHandler.handleServerStarting(FMLCommonHandler.java:278)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at net.minecraft.server.integrated.IntegratedServer.func_71197_b(IntegratedServer.java:94)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:387)
[05:15:39] [Server thread/INFO] [STDERR]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685)
[05:15:39] [Server thread/INFO] [ProjectE]: Collected Mappings from moze_intel.projecte.emc.mappers.FluidMapper
[05:15:39] [Server thread/INFO] [ProjectE]: Collected Mappings from moze_intel.projecte.emc.mappers.SmeltingMapper
[05:15:39] [Server thread/INFO] [ProjectE]: Collected Mappings from moze_intel.projecte.emc.mappers.APICustomConversionMapper
[05:15:39] [Server thread/INFO] [ProjectE]: Mapping Collection finished
[05:15:39] [Server thread/INFO] [ProjectE]: Starting to generate Values:
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*OD: gemEmerald => 1*minecraft:emerald(388:0)" ingredient cost: 8192/1 fixed value of result: 16384/1
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*ActuallyAdditions:itemDust(4423:6) => 1*minecraft:coal(263:0)" ingredient cost: 64/1 fixed value of result: 128/1
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*minecraft:rotten_flesh(367:0) => 1*minecraft:leather(334:0)" ingredient cost: 32/1 fixed value of result: 64/1
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*OD: gemPeridot => 1*essentialcraft:genItem(9087:40)" ingredient cost: 2048/1 fixed value of result: 8192/1
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*OD: gemSapphire => 1*essentialcraft:genItem(9087:40)" ingredient cost: 2048/1 fixed value of result: 8192/1
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*OD: gemRuby => 1*essentialcraft:genItem(9087:40)" ingredient cost: 2048/1 fixed value of result: 8192/1
[05:15:39] [Server thread/WARN] [ProjectE]: EMC Exploit: "0/1 + 1*DCsAppleMilk:defeatedcrow.dustWood(4753:1) => 1*OD: dustCoal" ingredient cost: 16/1 fixed value of result: 64/1
[05:15:39] [Server thread/INFO] [ProjectE]: Generated Values...
[05:15:39] [Server thread/INFO] [ProjectE]: Registered 977 EMC values. (took 481 ms)

I also got a similar error with Real Train Mod a Japanese mod.

**Comment by MaPePeR** [comment]
The Real Train Mod-Issue should be fixed by the Pull Request i created.

The `ArrayIndexOutOfBoundsException` in `ErebusFood.getContainerItem` is caused by us calling `getContainerItem` on an `ItemStack` with damage value = wildcard. We probably should not do that.

(But its also something they should fix in their code)

**Comment by accessdenied0** [comment]
Yeah, this is a possible reason like @luffykick said, that it could also a mapper issue with NEID, but I do not have the rest of the mods @luffykick have.

**Comment by accessdenied0** [comment]
Hey everyone,

I know this is closed, but I lately started a small Mod-pack and I run again in the issues that the Values of most of my items are gone. This time strangely enough it happens with Items like wooden-things and even Iron-based Items (even they crafting components have a EMC)

I post here the [log](https://drive.google.com/file/d/0B1BMWHu8n1hXX0lPeXpodXlGOWc/view?usp=sharing) and my [EMC-log](https://drive.google.com/file/d/0B1BMWHu8n1hXc0F5SWFvR2dJSkE/view?usp=sharing)

Because I report this again, it seem to be much worse than in my other modpack and even if I try to start a world it failed to do so and send me back to title screen. Only after re-trying it let me load the world up.

access_denied

**Comment by MaPePeR** [comment]
@accessdenied0 the #1133 PR should fix this issue.

Please try [this build](https://github.com/MaPePeR/ProjectE/releases/tag/v1.9.4-craftingmapperfixes) and see If it fixes your EMC values.

The "I need to load my world twice"-thing is not caused by ProjectE, i think. From your log:

```
[07:08:54] [Netty Client IO #1/INFO] [EnderCore/]: Reset configs to client values for com.enderio.core.common.config.ConfigHandler
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:176)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.Gson.fromJson(Gson.java:803)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.Gson.fromJson(Gson.java:768)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.Gson.fromJson(Gson.java:717)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.Gson.fromJson(Gson.java:689)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at kihira.tails.common.Tails.loadConfig(Tails.java:136)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at kihira.tails.client.ClientEventHandler.onDisconnect(ClientEventHandler.java:91)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.eventhandler.ASMEventHandler_501_ClientEventHandler_onDisconnect_ClientDisconnectionFromServerEvent.invoke(.dynamic)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:54)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.eventhandler.EventBus.post(EventBus.java:140)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at cpw.mods.fml.common.network.handshake.NetworkDispatcher.close(NetworkDispatcher.java:405)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at io.netty.channel.DefaultChannelHandlerContext.invokeClose(DefaultChannelHandlerContext.java:560)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at io.netty.channel.DefaultChannelHandlerContext.access$1200(DefaultChannelHandlerContext.java:28)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at io.netty.channel.DefaultChannelHandlerContext$14.run(DefaultChannelHandlerContext.java:550)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:354)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:348)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:101)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at java.lang.Thread.run(Unknown Source)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.stream.JsonReader.beginObject(JsonReader.java:374)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:165)
[07:08:54] [Netty Client IO #1/INFO] [STDERR/]: [java.lang.Throwable$WrappedPrintStream:println:-1]: 	... 17 more
[07:08:54] [Netty Client IO #1/INFO] [EnderCore/]: Reset configs to client values for tterrag.wailaplugins.config.WPConfigHandler
[07:08:54] [Netty Client IO #1/DEBUG] [OpenMods/]: openmods.network.IdSyncManager.onDisconnect(IdSyncManager.java:124): Disconnected, restoring local data
[07:08:54] [Client thread/INFO] [journeymap/]: Mapping halted in C:\Program Files\MultiMC\instances\Vanilla Plus\minecraft\journeymap\data\sp\test\DIM0
[07:08:54] [Server thread/TRACE] [mcp/mcp]: Sending event FMLServerStoppingEvent to mod mcp
[07:08:54] [Server thread/TRACE] [mcp/mcp]: Sent event FMLServerStoppingEvent to mod mcp
```

**Comment by accessdenied0** [comment]
Oh okay...

I used your build on my light mod and I'm happy to report that most of the EMC are back.

Sadly it seem not to fix the chisel ones and here an there some "ArithmeticException"-s. [I uploaded here the Forge-log.](https://drive.google.com/file/d/0B1BMWHu8n1hXTEptVy1fbmhVSzQ/view?usp=sharing)

Also to mention but not so important is that when I try to "B:dumpEverythingToFile=true" I get a GC overflow.

access_denied

**Comment by MaPePeR** [comment]
Try [this version](https://github.com/MaPePeR/ProjectE/releases/tag/v1.9.4-craftingmapperfixes2) then, please. This should fix the Chisel2Mapper from Crashing and provide more insight on why it was crashing before. (Normal Chisel 2 works fine for me, so i suspect its chiseltones or something like that)

**Comment by accessdenied0** [comment]
@MaPePeR,

Everything seem to work! It is very awesome! Thank you a lot for your effort. There just where [just one overflow: mul](http://pastebin.com/bz2P3RfR). But this is on my "light" modded world. I try to use this mod on my other instance very soon. But so far I can tell, It works perfectly fine and it would surprise me if it doesn't work.

If I find some of these Messages like (overflow: mul or ArithmeticException) should I post them here or on a new Issue? (Or are these not important at all?)

access_denied

**Comment by MaPePeR** [comment]
The overflows are not important. I should remove them, probably. 

It should not clear your knowledge, though. 

Thanks for testing.
